src/sentry/api/endpoints/project_integration_details.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import
+
+from sentry.api.bases.project import ProjectEndpoint, ProjectIntegrationsPermission
+from sentry.api.serializers import serialize
+from sentry.models import ProjectIntegration, Integration
+
+
+class ProjectIntegrationDetailsEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectIntegrationsPermission, )
+
+    def get(self, request, project, integration_id):
+        integration = ProjectIntegration.objects.get(
+            project=project,
+            integration_id=integration_id,
+        )
+
+        return self.respond(serialize(integration, request.user))
+
+    def put(self, request, project, integration_id):
+        # Integrations can only be added to a project if they are already
+        # configured for the organization themselves.
+        try:
+            integration = Integration.objects.get(
+                id=integration_id,
+                organizations=project.organization_id,
+            )
+        except Integration.DoesNotExist:
+            return self.respond(status=404)
+
+        created = integration.add_project(project.id)
+
+        return self.respond(status=(201 if created else 204))
+
+    def delete(self, request, project, integration_id):
```

Comment:
tbh, I think this is a perfect `DetailView` use case. https://ccbv.co.uk/projects/Django/1.3/django.views.generic.detail/DetailView/

Reply:
These aren't even Django CBV. They are from Django Rest Framework. We also use this pattern for every other API endpoint. We're not changing now.

Reply:
I understand `We're not changing now`, but if this is a DRF endpoint, then perfect it is a `RetrieveUpdateDestroyAPIView` use case. Forget the views, but the duplication in selecting the object seems a bit .. yeah - I see `We're not changing now`. :D